presto-iceberg/src/test/java/io/prestosql/plugin/iceberg/AbstractTestIcebergSmoke.java

```diff
@@ -1389,4 +1395,55 @@ private void dropTable(String table)
         assertUpdate(session, "DROP TABLE " + table);
         assertFalse(getQueryRunner().tableExists(session, table));
     }
+
+    @Test
+    public void testLocalDynamicFilterWithEmptyBuildSide()
```

Comment:
These tests rely on fact that probe side will not progress until build side is not ready. Because of that these tests are fragile and might break when we change execution (which we will).
In order to write these tests reliable, Iceberg should support either waiting for local DF or waiting for DF for split generation. Then (in tests) we can enforce presence of DF before table scan happens

Reply:
Could we tackle introduce waiting for split generation like Hive separately and let this one go through with less restrictive assertions ?
Interestingly, `TestHiveDistributedJoinQueries#testJoinWithEmptyBuildSide` is not flaky despite not using lazy DF to delay probe side.

Reply:
This would prevent optimization like https://github.com/prestosql/presto/issues/3957
I'm fine having local DF without testing for now (it's one liner), but please create an issue in prestosql (for blocking on DFs locally) and mention it in code.

Alternatively, we can simply add that blocking because it's pretty straight forward (just pass DF future to `IcebergPageSource` and block there)